mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1819,8 +1890,8 @@ void mlir::vector::populateShapeCastFoldingPatterns(RewritePatternSet &patterns,
 
 void mlir::vector::populateDropUnitDimWithShapeCastPatterns(
```

Comment:
These patterns are added to `populateFlattenVectorTransferPatterns`: https://github.com/llvm/llvm-project/blob/662b1305292d69507a09164c911cfc782e0e4e73/mlir/lib/Dialect/Vector/Transforms/VectorTransferOpTransforms.cpp#L955-L963

However, they don't "flatten transfer Ops" (sorry, I might have started this confusing trend). This makes:
*  [vector-transfer-flatten.mlir](https://github.com/llvm/llvm-project/pull/93007/files#diff-5b03f5f1c68b2b50103abe956116e1a251bcfed7fea8c11c1f3f15914d226f5b) 

a bit confusing. Let's fix this by:
* adding a dedicated test pass in "TestVectorTransforms.cpp" (*), and 
* moving tests for `populateDropUnitDimWithShapeCastPatterns ` to dedicated file. 

(*) Alternatively, could you add a new Op in "VectorTransformOps.td" for these patterns? Whatever requires less code :)